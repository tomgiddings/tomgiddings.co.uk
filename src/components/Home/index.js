// @flow
import React from 'react';
import Footer from '../Footer'

class Home extends React.Component<*> {
  state = {
    hour: null,
    date: new Date(),
    timeStamp: ''
  }

  componentDidMount() {
    this.getHour();
    this.getTimeStamp();
  }

  padZero = (value) => {
    return (value < 10 ? '0' + (value.toString()).slice(-2) : value);
  }

  getTimeStamp = () => {
    const days = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
    const months = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'];
    const timeStamp = days[this.state.date.getDay()] + " "
      + months[this.state.date.getMonth()] + ' '
      + this.state.date.getDate() +  ' '
      + this.padZero(this.state.date.getHours()) + ':'
      + this.padZero(this.state.date.getMinutes()) + ':'
      + this.padZero(this.state.date.getSeconds());
    this.setState({
      timeStamp
    });
  }

  getHour = () => {
   const date = new Date();
   const hour = this.state.date.getHours()
   this.setState({
      hour
   });
  } 

  render() {
    const {hour, timeStamp} = this.state;
    document.title = 'Tom Giddings - Web Developer and Broadcast Engineer';
    return (
      <article className="page home">
        <div className="content-wrapper">
          <div className="terminal">
            <div className="fakeMenu">
              <div className="fakeButtons fakeClose"></div>
              <div className="fakeButtons fakeMinimize"></div>
              <div className="fakeButtons fakeZoom"></div>
            </div>
            <div className="fakeScreen">
              <p>Last login: {timeStamp} on ttys000</p>
              <p className="type-text">$ Good {hour < 12 ? `morning` : hour < 16 ? 'afternoon' : `evening`}.</p>
              <p className="type-text">$ I'm Tom, a Web Developer &amp; Broadcast Engineer.&#9608;</p>
            </div>
          </div>
        </div>
      </article>
    );    
  }
}
export default Home;
