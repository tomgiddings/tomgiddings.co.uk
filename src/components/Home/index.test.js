import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Home from './';

afterEach(cleanup);

it('Renders matching snapshot', () => {
  const realDateNow = Date.now.bind(global.Date);
  const dateNowStub = jest.fn(() => 1530518207007);
  global.Date.now = dateNowStub;

  const { asFragment } = render(<Home />);
  expect(asFragment()).toMatchSnapshot();
});

it('Correctly creates a last login string', () => {
  const dateNowStub = jest.fn(() => new Date("January 09, 2020 01:02:03"));
  global.Date.now = dateNowStub;
  const { getByText } = render(<Home />);
  expect(getByText('Last login: Thu Jan 9 01:02:03 on ttys000')).toBeDefined();
});

it('Says good morning', () => {
  const dateNowStub = jest.fn(() => new Date("January 09, 2020 01:02:03"));
  global.Date.now = dateNowStub;
  const { getByText } = render(<Home />);
  expect(getByText('$ Good morning.')).toBeDefined();
});

it('Says good afternoon', () => {
  const dateNowStub = jest.fn(() => new Date("January 09, 2020 12:02:03"));
  global.Date.now = dateNowStub;
  const { getByText } = render(<Home />);
  expect(getByText('$ Good afternoon.')).toBeDefined();
});

it('Says good evening', () => {
  const dateNowStub = jest.fn(() => new Date("January 09, 2020 16:02:03"));
  global.Date.now = dateNowStub;
  const { getByText } = render(<Home />);
  expect(getByText('$ Good evening.')).toBeDefined();
});
