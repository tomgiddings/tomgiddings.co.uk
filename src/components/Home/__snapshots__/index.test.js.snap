// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders matching snapshot 1`] = `
<DocumentFragment>
  <article
    class="page home"
  >
    <div
      class="content-wrapper"
    >
      <div
        class="terminal"
      >
        <div
          class="fakeMenu"
        >
          <div
            class="fakeButtons fakeClose"
          />
          <div
            class="fakeButtons fakeMinimize"
          />
          <div
            class="fakeButtons fakeZoom"
          />
        </div>
        <div
          class="fakeScreen"
        >
          <p>
            Last login: Mon Jul 2 08:56:47 on ttys000
          </p>
          <p
            class="type-text"
          >
            $ Good morning.
          </p>
          <p
            class="type-text"
          >
            $ I'm Tom, a Web Developer & Broadcast Engineer.
          </p>
          <p
            class="type-text"
          >
            $ Find out more using the menu in the top right.â–ˆ
          </p>
        </div>
      </div>
      <div
        class="buttons"
      >
        <h4>
          Find me online
        </h4>
        <a
          href="https://www.instagram.com/tomgiddings/"
        >
          <i
            aria-hidden="true"
            class="fab fa-instagram fa-3x"
            title="Instagram"
          />
          <span
            class="sr-only"
          >
            Instagram
          </span>
        </a>
        <a
          href="https://www.linkedin.com/in/tomgiddings/"
        >
          <i
            aria-hidden="true"
            class="fab fa-linkedin-in fa-3x"
            title="LinkedIn"
          />
          <span
            class="sr-only"
          >
            LinkedIn
          </span>
        </a>
        <a
          href="https://github.com/tomgiddings"
        >
          <i
            aria-hidden="true"
            class="fab fa-github fa-3x"
            title="GitHub"
          />
          <span
            class="sr-only"
          >
            GitHub
          </span>
        </a>
      </div>
    </div>
  </article>
</DocumentFragment>
`;
